libhttpserver/server.go

```diff
@@ -187,26 +188,46 @@ const portStart = 7000
 const portEnd = 8000
 const requestPathRoot = "/files/"
 
+func (s *Server) startServer() (err error) {
+	s.server = libkb.NewHTTPSrv(
+		s.g, libkb.NewPortRangeListenerSource(portStart, portEnd))
+	// Have to start this first to populate the ServeMux object.
+	if err = s.server.Start(); err != nil {
+		return err
+	}
+	s.server.Handle(requestPathRoot,
+		http.StripPrefix(requestPathRoot, http.HandlerFunc(s.serve)))
+	return nil
+}
+
+func (s *Server) monitorAppState() {
+	state := keybase1.AppState_FOREGROUND
+	for {
+		state = <-s.g.AppState.NextUpdate(&state)
```

Comment:
On mobile will you be sharing the same `AppState` as the service?  Does that mean only one of you will get this update?  It looks like `AppState.NextUpdate()` takes a lock, so there can only be one caller at once.  Maybe this doesn't share a `g` with the service?

Reply:
Oh I see, it actually doesn't block, and saves all the channels.  Ok nevermind.

Reply:
That’s actually a great point. I was rushing this in without thinking about that. Thanks for making sure!